ocaml/testsuite/tests/typing-layouts/annots.ml

```diff
@@ -103,20 +171,28 @@ type t = string t2
 type ('a : immediate) t2_imm
 type (_ : immediate) t2_imm'
 type t1 = int t2_imm
-type t2' = bool t2_imm
+type t2' = bool t2_imm'
 type ('a : float64) t2_float64
 type (_ : float64) t2_float64'
 type t3 = float# t2_float64
+type ('a : value mod global) t2_global
+type (_ : value mod global) t2_global'
+type ('a : word mod external_ many shared) t2_complex
+type (_ : word mod external_ many shared) t2_complex'
```

Comment:
Maybe this should be a warning, `mod`ding by a top.

Reply:
Based on our conversation, this is going to be left to do in another PR.